design-proposals/003-credentials.md

```diff
@@ -0,0 +1,81 @@
+# Credential support
+
+* Issue: https://github.com/syndesisio/syndesis-rest/issues/386
+* Sprint: 13
+* Affected Repos:
+  - syndesis-rest
+  - syndesis-ui
+  
+## Background
+
+High level goals are:
+
+ * Help users provide and manage credentials
+ * Persist credentials in a secure manner
+ * Apply credentials to connections with minimum impact on the existing implementation
+
+## User Story
+
+As a citizen user I would like to connect to 3rd party service that require authenticate using credentials. Credentials would need to be stored in Syndesis so I can manage and assign them to other integrations using same 3rd party services. For example when connecting to Salesforce the I need to provide username/password authentication only once to authorize Syndesis's access using OAuth, and in order to do that I would like to choose "connect to Salesforce" in Syndesis and not be prompted again for authorization.
```

Comment:
No I don't think same credentials will be reused on different connections. I need to clarify that.